src/ngx_http_lua_balancer.c

```diff
@@ -537,6 +537,72 @@ ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
 
 
 int
+ngx_http_lua_ffi_balancer_set_peer_timeout(ngx_http_request_t *r,
+    const u_char *type, int timeout, char **err)
```

Comment:
Instead of using `int` for the timeout argument, I think `long` is better, which is usually larger than `int` on 64-bit systems. And timeout values could be large.